catalog/data_metastores.go

```diff
@@ -0,0 +1,25 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/unitycatalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceMetastores() *schema.Resource {
+	type MetastoresData struct {
+		Metastores []unitycatalog.MetastoreInfo `json:"metastores,omitempty" tf:"computed"`
```

Comment:
I've seen indeed that this is something that is done with other data sources... but for me this is not really nice for the user, since they would need two resource blocks to get the information of my metastores (first the `data_metastores`, then the `data_metastore` passing the specific ID of the metastore).